sdk/src/main/java/com/mapbox/maps/renderer/MapboxWidgetRenderer.kt

```diff
@@ -0,0 +1,128 @@
+package com.mapbox.maps.renderer
+
+import android.opengl.GLES20
+import androidx.annotation.VisibleForTesting
+import com.mapbox.common.Logger
+import com.mapbox.maps.renderer.egl.EGLCore
+import java.util.concurrent.CopyOnWriteArraySet
+import javax.microedition.khronos.egl.EGL10
+import javax.microedition.khronos.egl.EGLContext
+import javax.microedition.khronos.egl.EGLSurface
+
+internal class MapboxWidgetRenderer {
+
+  companion object {
+    private const val TAG: String = "MapboxWidgetRenderer"
+  }
+
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  internal var widgetEglPrepared = false
+  private var widgetEglSurface: EGLSurface = EGL10.EGL_NO_SURFACE
+  private lateinit var widgetEglCore: EGLCore
+
+  private val textures = intArrayOf(0)
+  private val framebuffers = intArrayOf(0)
+
+  private val widgets = CopyOnWriteArraySet<BaseWidget>()
+
+  fun getTextureId() = textures[0]
+
+  fun prepare(
+    sharedContext: EGLContext, width: Int, height: Int
+  ): Boolean {
+    widgetEglCore = EGLCore(
+      translucentSurface = true,
+      antialiasingSampleCount = 1,
+      sharedContext = sharedContext,
+    )
+
+    if (!widgetEglPrepared) {
+      widgetEglCore.prepareEgl()
+      if (widgetEglCore.eglStatusSuccess) {
+        widgetEglPrepared = true
+      } else {
+        Logger.e(TAG, "Widget EGL was not configured, please check logs above.")
+        return false
+      }
+    }
+    widgetEglSurface = widgetEglCore.createOffscreenSurface(width = width, height = height) ?: EGL10.EGL_NO_SURFACE
+    if (!widgetEglCore.eglStatusSuccess) {
+      // Set EGL Surface as EGL_NO_SURFACE and try recreate it in next iteration.
+      widgetEglSurface = EGL10.EGL_NO_SURFACE
+      return false
+    }
+
+    createTexture(width, height)
+
+    return true
+  }
+
+  private fun createTexture(width: Int, height: Int) {
```

Comment:
I'd just put it all in `prepare()` or change naming as it's preparing not only texture but more an associated framebuffer